### PR TITLE
Fix: Swagger UI shows obscure error for non-fetch file and manifest endpoints (#3599)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**
Connected issue: #3599 

### Author

- [x] PR title references the connected issue
- [x] PR title matches<sup>1</sup> title of connected issue         <sub>or comment in PR explains why they're different</sub>
- [x] Title of at least one commit references connected issue
- [x] PR is connected to issue via Zenhub 
- [x] PR description links to connected issue

<sup>1</sup> when the issue title describes a problem, the PR title is `Fix: ` followed by the issue title   

### Author (reindex)

- [x] Added `r` tag to commit title                                 <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                                   <sub>or this PR does not require reindexing</sub>

### Author (chains)

- [x] This PR is blocked by previous PR in the chain                <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                        <sub>or this PR is not chained to another PR</sub>

### Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst          <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                                 <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                                   <sub>or this PR does not require upgrading</sub>
- [x] Added announcement to PR description                          <sub>or this PR does not require announcement</sub>
- [x] Added checklist items for additional operator tasks           <sub>or this PR does not require additional tasks</sub>

## Author (hotfixes)

- [x] Added `h` tag to commit title and PR targets `prod`           <sub>or this PR does not include a temporary hotfix</sub>
- [x] Added `H` tag to commit title                                 <sub>or this PR does not include a permanent hotfix</sub>
- [x] Added `hotfix` label to PR                                    <sub>or this PR does not include a hotfix</sub>
- [x] Reverted the temporary hotfix connected to the issue          <sub>or there is no temporary hotfix for the connected issue on the `prod` branch</sub>

### Author (requirements, before every review)

- [x] Ran `make requirements_update`                                <sub>or this PR does not touch requirements*.txt, common.mk, Makefile and Dockerfile</sub>
- [x] Added `R` tag to commit title                                 <sub>or this PR does not touch requirements*.txt</sub>
- [x] Added `reqs` label to PR                                      <sub>or this PR does not touch requirements*.txt</sub>

### Author (before every review)

- [x] `make integration_test` passes in personal deployment         <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

### Primary reviewer (after approval)

- [x] Labeled connected issue as `demo` or `no demo`
- [x] Commented on connected issue about demo expectations          <sub>or labelled connected issue as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

### Operator (before pushing merge the commit)

- [x] Checked `reindex` label and `r` commit title tag
- [x] Checked that demo expectations are clear                      <sub>or connected issue is labeled as `no demo`</sub>
- [x] Rebased and squashed branch
- [x] Sanity-checked history
- [x] Pushed PR branch to Github
- [x] Branch pushed to Gitlab and added `sandbox` label             <sub>or PR is labeled `no sandbox`</sub>
- [x] Build passed in sandbox                                       <sub>or PR is labeled `no sandbox`</sub>
- [x] Started reindex in `sandbox`                                  <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Checked for failures in `sandbox`                             <sub>or this PR does not require reindexing `sandbox`</sub>
- [x] Added PR reference to merge commit title
- [x] Collected commit title tags in merge commit title
- [x] Moved connected issue to Merged column
- [x] Pushed merge commit to Github

### Operator (after pushing the merge commit)

- [x] Made announcement requested by author                         <sub>or PR description does not contain an announcement</sub>
- [ ] Shortened the PR chain                                        <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate               <sub>or this PR is authored by lead</sub>
- [ ] Pushed merge commit to Gitlab                                 <sub>or merge commit can be pushed later, with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab
- [ ] Build passes on Gitlab
- [ ] Moved connected issue to `prod` or `Merged prod`              <sub>or this PR does not represent a promotion</sub>

### Operator (reindex) 

- [ ] Started reindex in target deployment                          <sub>or this PR does not require reindexing</sub>
- [ ] Checked for and triaged indexing failures                     <sub>or this PR does not require reindexing</sub>
- [ ] Emptied fail queues in target deployment                      <sub>or this PR does not require reindexing</sub>
- [ ] Filed backport PR                                             <sub>or this PR does not represent a hotfix</sub>

### Operator

- [ ] Unassigned PR

## Shorthand for review comments

- `L` line is too long
- `W` line wrapping is wrong
- `Q` bad quotes
- `F` other formatting problem

### CODE REVIEW

**Review by danielsotirhos — APPROVED** [review]
Nicely done.

File: lambdas/service/app.py [hunk 1]
```diff
@@ -1476,62 +1480,74 @@ def manifest_path_spec(*, fetch: bool):
...
+        'description': format_description('''
+            Initiate and check status of a manifest generation job, returning
+            either a 301 response redirecting to a URL to re-check the status of
+            the manifest generation or a 302 response redirecting to the location
+            of the completed manifest.
+
+            This endpoint is not suitable for interactive use via a web browser due
+            to the maximum number of retries being rapidly exhausted. Use the
+            [/fetch endpoint](#operations-Manifests-get_fetch_manifest_files)
+            instead.
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:1501]:
```suggestion
            This endpoint is not suitable for interactive use via Swagger UI. Please use the
            [/fetch endpoint](#operations-Manifests-get_fetch_manifest_files)
            instead.
```

The explanation is an implementation detail, and your rendition of it is incomplete.

File: lambdas/service/vendor/resources/static/swagger-ui.html.template.mustache [hunk 2]
```diff
@@ -32,6 +32,19 @@
     <script src="https://cdnjs.cloudflare.com/ajax/libs/swagger-ui/3.24.2/swagger-ui-standalone-preset.js" integrity="sha384-sTb47WR//agLnr7p+2eMNLxtfLZFYwBzMyh4U/NMCrBeDjPe6leIOcU+7IgdvnBt" crossorigin="anonymous"></script>
     <script>
     window.onload = function() {
+      const DisableTryItOutPlugin = function() {
```
Comment (hannes-ucsc) [inline lambdas/service/vendor/resources/static/swagger-ui.html.template.mustache:35]: Assuming you took this off the web, you should attribute the source.

File: src/azul/chalice.py [hunk 3]
```diff
@@ -67,6 +69,7 @@ def __init__(self,
         assert app_module_path.endswith('/app.py'), app_module_path
         self.app_module_path = app_module_path
         self.unit_test = unit_test
+        self.non_interactive_methods: Set[Tuple[str, str]] = set()
```
Comment (hannes-ucsc) [inline src/azul/chalice.py:72]:
```suggestion
        self.non_interactive_routes: Set[Tuple[str, str]] = set()
```

and so on

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

File: lambdas/service/app.py [hunk 4]
```diff
@@ -1476,62 +1480,73 @@ def manifest_path_spec(*, fetch: bool):
...
+        'tags': ['Manifests'],
+        'summary': 'Request a download link to a manifest file and redirect',
+        'description': format_description('''
+            Initiate and check status of a manifest generation job, returning
+            either a 301 response redirecting to a URL to re-check the status of
+            the manifest generation or a 302 response redirecting to the location
+            of the completed manifest.
+
+            This endpoint is not suitable for interactive use via the Swagger UI.
+            Please Use the [/fetch endpoint](#operations-Manifests-get_fetch_manifest_files)
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:1499]:
```suggestion
            Please use [/fetch endpoint](#operations-Manifests-get_fetch_manifest_files)
```

File: lambdas/service/app.py [hunk 5]
```diff
@@ -1687,10 +1702,15 @@ def generate_manifest(event: AnyJSON, context: LambdaContext):
...
 
 
-@app.route('/repository/files/{file_uuid}', methods=['GET'], cors=True, method_spec={
+@app.route('/repository/files/{file_uuid}', methods=['GET'], interactive=False, cors=True, method_spec={
     **repository_files_spec,
     'summary': 'Redirect to a URL for downloading a given data file from the '
                'underlying repository',
+    'description': format_description('''
+        This endpoint is not suitable for interactive use via the Swagger UI.
+        Please use the [/fetch endpoint](#operations-Repository-get_fetch_repository_files__file_uuid_)
```
Comment (hannes-ucsc) [inline lambdas/service/app.py:1711]:
```suggestion
        Please use [/fetch endpoint](#operations-Repository-get_fetch_repository_files__file_uuid_)
```

**Review by hannes-ucsc — CHANGES_REQUESTED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Pull request merged**